metadata/src/episode.rs

```diff
@@ -36,7 +33,7 @@ pub struct Episode {
     pub covers: Images,
     pub language: String,
     pub is_explicit: bool,
-    pub show: SpotifyId,
+    pub show: String,
```

Comment:
Similarly to https://github.com/librespot-org/librespot/pull/1018, more metadata could be made available here.

```suggestion
    pub show: Show,
```

Reply:
That was my thought too but if you do:
```rust
pub show: Show,
```

it fails 100% of the time. I'm not sure why?

And if you pretty print (`{:#?}`) episode all you get for show is an id and name:
```
gid: "\002\236k\322\225?S\334\253\244\263s\004\307\332\034"
name: "Has America Learned Anything From Its Biggest Disasters? Not Really."
duration: 2258964
audio {
  file_id: "*\356$:\202\336\355*\025\343\373\\\366n\257\303\347\363\201F"
  format: AAC_24
}
audio {
  file_id: "\222\356q\254]\310\256\335[\367\035\347\376\3760W\230/\260p"
}
audio {
  file_id: "\035\243Qv\3779\310\333+y\247\2274\025ImrI?B"
}
audio {
  file_id: "zl\322\2312\010\334\346\351\375\263\3313\266$|\372\245\2010"
  format: OGG_VORBIS_96
}
description: "Jon talks to Pulitzer Prize-winning journalist Sheri Fink, whose book Five Days at Memorial is the basis for the Apple TV+ series of the same name. They discuss why our healthcare system is always ill-equipped for disasters\342\200\224and whether there\342\200\231s something else coming our way that we should panic about. Writers Robby Slowik and Maria Randazzo also stop by to talk about the Trump raid, the oddest menu items at Mar-a-Lago, and the scourge of lanternfliesCREDITSHosted by: Jon StewartFeaturing, in order of appearance: Robby Slowik, Maria Randazzo, Sheri FinkExecutive Produced by Jon Stewart, Brinda Adhikari, James Dixon, Chris McShane, and Richard Plepler.Lead Producer: Sophie EricksonProducers: Zach Goldbaum, Caity Gray, and Robby SlowikAssoc. Producer: Andrea BetanzosSound Engineer & Editor: Miguel CarrascalSenior Digital Producer: Frederika MorganDigital Coordinator: Norma HernandezSupervising Producer: Lorrie BaranekHead Writer: Kris AcimovicElements: Kenneth Hull, Daniella PhilipsonTalent: Brittany Mehmedovic, Marjorie McCurry, Lukas Thimm\302\240Research: Susan Helvenston, Andy Crystal, and Cassie MurdochTheme Music by: Gary Clark Jr.The Problem with Jon Stewart podcast is an Apple TV+ podcast produced by Busboy Productions.https://apple.co/-JonStewart"
publish_time {
  year: 2022
  month: 8
  day: 25
  hour: 10
  minute: 0
}
cover_image {
  image {
    file_id: "\253gec\000\000\366\215\254\304E\013!\277nBJ\356R\370"
    size: SMALL
    width: 64
    height: 64
  }
  image {
    file_id: "\253gec\000\000_\037\254\304E\013!\277nBJ\356R\370"
    size: DEFAULT
    width: 300
    height: 300
  }
  image {
    file_id: "\253gec\000\000\272\212\254\304E\013!\277nBJ\356R\370"
    size: LARGE
    width: 640
    height: 640
  }
}
language: "en"
explicit: true
show {
  gid: "\017\037\312\321\333\035C\334\223\224\210q-\361\322J"
  name: "The Problem With Jon Stewart"
}
audio_preview {
  file_id: "\351\275(\227w\214\224\244\376\016+\003\017\236\327N\213w\177\216"
  format: MP3_96
}
restriction {
  catalogue_str: "all"
  catalogue_str: "free"
  catalogue_str: "premium"
  catalogue_str: "shuffle"
  catalogue_str: "commercial"
  countries_forbidden: ""
}
allow_background_playback: false
external_url: "https://rss.art19.com/episodes/448a50e8-b4c6-4af5-91ab-67783737bca7.mp3?rss_browser=BAhJIgxTcG90aWZ5BjoGRVQ%3D--c73337271af676556b67a4f8fe9332bd11c6baf9"
type: FULL
```

Reply:
I'm not exactly sure where these metadata structs came form but a lot of them are wrong or missing?

Reply:
I haven't yet looked into it properly, but my guess is that it tries to deserialize `show.trailer_uri`, which is a `SpotifyId`, from an empty string, since the field is not supplied. This fails, and thus the creation of the whole struct errors.

The general problem is that the metadata structs in the `.proto` files are used in many places, while the way in which they are filled varies widely. For example, if you request a show's metadata via the endpoint `/metadata/4/show/<id>`, you'll receive a much more complete `Show` struct, while in other instances, such as this one, only very few fields are actually supplied.

There were already some discussions about that issue in multiple places, but surely out of scope for this PR. :)

Reply:
It compiles just fine it just makes it so retrieving the episode fails like it's somehow clobbering the episode's id?

Reply:
> I'm not exactly sure where these metadata structs came from but a lot of them are wrong or missing?

When I pretty much rewrote the entire `metadata` package I guessed as good as I could from the protobuf definitions. I did not meticulously verify everything and no one else bothered to test. So I am glad this is getting some showtime now.

> I haven't yet looked into it properly, but my guess is that it tries to deserialize `show.trailer_uri`, which is a `SpotifyId`, from an empty string, since the field is not supplied. This fails, and thus the creation of the whole struct errors.

It's been a while, but if that's the case then I'd consider it a bug in need of fixing.

Reply:
>  For example, if you request a show's metadata via the endpoint /metadata/4/show/<id>, you'll receive a much more complete Show struct, while in other instances, such as this one, only very few fields are actually supplied.

When it comes to these events I'm only really concerned with what we can get for free (and that actually works, or at least doesn't cause failures) We're not making other calls to get the info in AudioItem.

Reply:
I agree.